compiler/rustc_hir_typeck/src/expr_use_visitor.rs

```diff
@@ -438,6 +438,12 @@ impl<'a, 'tcx> ExprUseVisitor<'a, 'tcx> {
                         // to borrow discr.
                         needs_to_be_read = true;
                     }
+                    PatKind::Slice(&[], Some(_), &[]) if !discr_place.place.base_ty.is_array() => {
+                        // If the pattern is a wildcard only, we don't need to read the length
+                    }
+                    PatKind::Slice(..) if !discr_place.place.base_ty.is_array() => {
+                        needs_to_be_read = true;
+                    }
                     PatKind::Or(_)
                     | PatKind::Box(_)
                     | PatKind::Slice(..)
```

Comment:
Can this fallback be removed?

Reply:
@clubby789 - you marked @cjgillot's suggestions as resolved, but it'd be helpful to see a quick comment about why. It looks like these are mainly code organization suggestions rather than a functional change, so maybe you decided you preferred the way it was written at first?

Reply:
Oh my bad 😓 I didn't notice that I hadn't saved the file in my IDE before testing and pushed it. I'll push the changes now